src/main/java/edu/wpi/grip/ui/controllers/MainWindowController.java

```
@@ -1,9 +1,134 @@
+            "    return a * b\n"
+    );
+
+    @Override
+    public void initialize(URL url, ResourceBundle resourceBundle) {
+        PaletteView paletteView = new PaletteView(eventBus);
+        paletteView.operationsProperty().addAll(this.add, this.multiply);
+        this.topPane.getItems().add(paletteView);
+
+        PipelineView pipelineView = new PipelineView(eventBus, new Pipeline(this.eventBus));
+        this.bottomPane.setContent(pipelineView);
+
+        /* Add and remove some steps and connections */
+        new Thread(() -> {
+            try {
+                Thread.sleep(1000);
+
+                Step step1 = new Step(eventBus, add);
+                eventBus.post(new StepAddedEvent(step1));
+
+                Thread.sleep(1000);
+
+                Step step2 = new Step(eventBus, multiply);
+                eventBus.post(new StepAddedEvent(step2));
+
+                Thread.sleep(1000);
+
+                Step step3 = new Step(eventBus, add);
+                eventBus.post(new StepAddedEvent(step3));
+
+                Thread.sleep(1000);
+
+                @SuppressWarnings("unchecked")
+                Connection<Number> connection1 = new Connection<>(eventBus, (Socket<Number>) step1.getOutputSockets()[0],
+                        (Socket<Number>) step2.getInputSockets()[0]);
+                eventBus.post(new ConnectionAddedEvent(connection1));
+
+                Thread.sleep(1000);
+
+                @SuppressWarnings("unchecked")
+                Connection<Number> connection2 = new Connection<>(eventBus, (Socket<Number>) step2.getOutputSockets()[0],
+                        (Socket<Number>) step3.getInputSockets()[1]);
+                eventBus.post(new ConnectionAddedEvent(connection2));
+
+                Thread.sleep(1000);
+
+                eventBus.post(new ConnectionRemovedEvent(connection1));
+
+                Thread.sleep(1000);
+
+                eventBus.post(new StepRemovedEvent(step3));
+            } catch (Exception e) {}
```

Probably shouldn't silently catch the exception. It should make some noise.